lib/katalyst/basic/auth/middleware.rb

```diff
@@ -33,6 +22,7 @@ def username
 
         def default_username
           if Object.const_defined?("Rails")
+            # this is only needed for Rails 3 compatibility
```

Comment:
consider moving defaults to `rails.rb` or `auth.rb` – I think the middleware should be responsible for checking auth but does not need to define defaults

Reply:
I've moved these methods to a Config class